### PR TITLE
feat: multiple galleries handling, chore: ESLint linting & prettier

### PR DESCRIPTION
**Pull request by natterstefan**
## Notice

This branch will be merged into `next` and published in our _internal_ npm repo. _This is not published on npm_.

Due to time constraints for our company project, we had to move on and publish this internally already.

Follow the PR in the forked repo instead: https://github.com/xiaolin/react-image-gallery/pull/446.

## Most important code changes for this feature

(FTR: chronological order is important)

- https://github.com/lovelysystems/react-image-gallery/pull/1/files?w=1#diff-eabaad0cc8751d0947757ba5349ae171R939-R958
- https://github.com/lovelysystems/react-image-gallery/pull/1/files?w=1#diff-eabaad0cc8751d0947757ba5349ae171R1058-R1086
- https://github.com/lovelysystems/react-image-gallery/pull/1/files?w=1#diff-eabaad0cc8751d0947757ba5349ae171R1088-R1111
- https://github.com/lovelysystems/react-image-gallery/pull/1/commits/7b23c2d2579ecc27194f88f9fa3e3459c7e99b41
- https://github.com/lovelysystems/react-image-gallery/pull/1/commits/df8334fb2f09549b159792d355f5a1b3b73f45a2

## Related PR

- Original Repo: https://github.com/xiaolin/react-image-gallery/pull/446

**Hint:** [enable "no whitespace" when reviewing](https://github.com/lovelysystems/react-image-gallery/pull/1/files?w=1), as a lot of ESLint things crossed my way ...

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+10.16
```
Comment (barnabasJ) [inline .nvmrc:1]: ch-js is on 12.11.1 should we use the same version here?

**Review by barnabasJ — APPROVED** [review]

File: .nvmrc (hunk 1 above)
Comment (natterstefan) [reply .nvmrc:1]: as discussed ;)

File: example/app.js [hunk 2]
```diff
@@ -214,131 +222,209 @@ class App extends React.Component {
...
+                <div className="app-interval-input-group">
+                  <span className="app-interval-label">Play Interval</span>
                   <input
-                    className='app-interval-input'
-                    type='text'
-                    onChange={this._handleInputChange.bind(this, 'slideInterval')}
-                    value={this.state.slideInterval}/>
+                    className="app-interval-input"
+                    type="text"
+                    onChange={this._handleInputChange.bind(
```
Comment (lukasender) [inline example/app.js:247]: is it really necessary to bind this on every `render`?

File: example/app.js [hunk 3]
```diff
@@ -214,131 +222,209 @@ class App extends React.Component {
...
+                <div className="app-interval-input-group">
+                  <span className="app-interval-label">Slide Duration</span>
                   <input
-                    className='app-interval-input'
-                    type='text'
-                    onChange={this._handleInputChange.bind(this, 'slideDuration')}
-                    value={this.state.slideDuration}/>
+                    className="app-interval-input"
+                    type="text"
+                    onChange={this._handleInputChange.bind(
```
Comment (lukasender) [inline example/app.js:262]: same question here

File: example/app.js [hunk 4]
```diff
@@ -214,131 +222,209 @@ class App extends React.Component {
...
-                  <span className='app-interval-label'>Thumbnail Bar Position</span>
+                <div className="app-interval-input-group">
+                  <span className="app-interval-label">
+                    Thumbnail Bar Position
+                  </span>
                   <select
-                    className='app-interval-input'
+                    className="app-interval-input"
                     value={this.state.thumbnailPosition}
                     onChange={this._handleThumbnailPositionChange.bind(this)}
```
Comment (lukasender) [inline example/app.js]: another one 🤔

File: example/app.js [hunk 5]
```diff
@@ -214,131 +222,209 @@ class App extends React.Component {
...
 
-            <ul className='app-checkboxes'>
+            <ul className="app-checkboxes">
               <li>
                 <input
-                  id='infinite'
-                  type='checkbox'
+                  id="infinite"
+                  type="checkbox"
                   onChange={this._handleCheckboxChange.bind(this, 'infinite')}
```
Comment (lukasender) [inline example/app.js:295]: bind

File: src/ImageGallery.jsx [hunk 6]
```diff
@@ -821,12 +908,14 @@ export default class ImageGallery extends React.Component {
...
-    this.resizeObserver = new ResizeObserver(debounce((entries) => {
-      if (!entries) return;
-      entries.forEach(() => {
-        this.handleResize();
-      });
-    }, 300));
+    this.resizeObserver = new ResizeObserver(
+      debounce(entries => {
+        if (!entries) return;
+        entries.forEach(() => {
```
Comment (lukasender) [inline src/ImageGallery.jsx]:
```suggestion
        entries.forEach(this.handleResize)
```

File: src/ImageGallery.jsx [hunk 7]
```diff
@@ -847,19 +936,26 @@ export default class ImageGallery extends React.Component {
...
-
-    if (onScreenChange) onScreenChange(fullScreenElement);
-    this.setState({ isFullscreen: !!fullScreenElement });
+    const fullScreenElement = ImageGallery.getFullScreenElement();
+    const isFullscreen = this.imageGallery.current === fullScreenElement;
+
+    this.setState({ isFullscreen }, () => {
+      if (onScreenChange) {
+        onScreenChange(
+          this.state.isFullscreen && fullScreenElement
```
Comment (barnabasJ) [inline src/ImageGallery.jsx]: is the && fullScreenElement necessary, isn't it already either the element or undefined?
```suggestion
          this.state.isFullscreen ? fullSreenElement : undefined
```

**Review by lukasender — APPROVED** [review]
found a couple of things but might be better to fix them in a separate pull request.

in `example/app.js`, I've found several more occurrences of `this.funcName.bind` than the one I've marked. Even though this might work, I think it's bad practice to use them as such.
And even though it's a forked example, I don't think we should continue to show bad practice.

Also, not a big thing, just something I noticed. In other projects we aim to use Object destructuring. Here, a lot of `this.state.xyz` is used.

Consider my comments, otherwise LGTM 👍 🚀

File: src/ImageGallery.jsx (hunk 7 above)
Comment (natterstefan) [reply src/ImageGallery.jsx]: Resolved here: https://github.com/lovelysystems/react-image-gallery/pull/1/commits/7b23c2d2579ecc27194f88f9fa3e3459c7e99b41#diff-eabaad0cc8751d0947757ba5349ae171R956-R968

File: example/app.js (hunk 2 above)
Comment (natterstefan) [reply example/app.js:247]: Will be handled later, for now I partially applied your feedback: https://github.com/lovelysystems/react-image-gallery/pull/1/commits/c9cdfee7a41d78fe41d670760347f66b1be00792.

File: example/app.js (hunk 3 above)
Comment (natterstefan) [reply example/app.js:262]: see https://github.com/lovelysystems/react-image-gallery/pull/1#discussion_r356095915

File: example/app.js (hunk 5 above)
Comment (natterstefan) [reply example/app.js:295]: see https://github.com/lovelysystems/react-image-gallery/pull/1#discussion_r356095915

**Review by lukasender — APPROVED** [review]

File: src/ImageGallery.jsx [hunk 8]
```diff
@@ -360,11 +235,21 @@ export default class ImageGallery extends React.Component {
...
+    // update both isFullscreen _and_ modalFullscreen
+    this.setState(
+      {
+        isFullscreen: state,
+        modalFullscreen: state,
+      },
+      () => {
+        // manually call because browser does not support screenchange events
+        if (onScreenChange) {
+          onScreenChange(state, prevIsFullscreen);
```
Comment (barnabasJ) [inline src/ImageGallery.jsx]:
```suggestion
          onScreenChange(state ? this.imageGallery.current : null, prevIsFullscreen);
```

**Review by barnabasJ — APPROVED** [review]

**Pull request merged**